backend/scripts/mirror_s3_data.sh

```diff
@@ -11,30 +11,80 @@ set -e
-set +x
+echo Copying WMG cube snapshot...
+
+AWS_OPTIONS="--copy-props metadata-directive --no-progress"
+S3_COPY_CMD="aws s3 cp $AWS_OPTIONS"
+S3_SYNC_CMD="aws s3 sync $AWS_OPTIONS"
+
+latest_snapshot=$(aws s3 cp s3://cellxgene-wmg-${SRC_ENV}/latest_snapshot_run -)
+if [[ $DEST_ENV == 'rdev' ]]; then
+  s3_destination="env-rdev-wmg/${STACK}"
+else
+  s3_destination="cellxgene-wmg-${DEST_ENV}"
+fi
+echo -n $latest_snapshot | aws s3 cp - s3://${s3_destination}/latest_snapshot_run
+$S3_SYNC_CMD s3://cellxgene-wmg-${SRC_ENV}/${latest_snapshot} s3://${s3_destination}/${latest_snapshot}
+
+
+echo Mirroring S3 Dataset data from $SRC_ENV to $DEST_ENV...
+
+if [[ $DEST_ENV == 'rdev' ]]; then
+  echo "Copying S3 Dataset assets for ${#COLLECTIONS[@]} Collections..."
+  DB_PW=$(aws secretsmanager get-secret-value --secret-id corpora/backend/${DEPLOYMENT_STAGE}/database --region us-west-2 | jq -r '.SecretString | match(":([^:]*)@").captures[0].string')
+
+  # Copy artifacts
+  uri_query="select uri from \"DatasetArtifact\" where id in (select unnest(artifacts) from \"DatasetVersion\" where id in (select unnest(datasets) from \"CollectionVersion\" where collection_id in ('$(sed "s/,/','/g" <<< $COLLECTIONS)')))"
+  DEPLOYMENT_STAGE=$DEPLOYMENT_STAGE make db/tunnel/up
+  uris=($(PGOPTIONS='-csearch_path=persistence_schema' PGPASSWORD=${DB_PW} psql --dbname corpora_${DEPLOYMENT_STAGE} --username corpora_${DEPLOYMENT_STAGE} --host 0.0.0.0 --csv --tuples-only -c "$uri_query"))
+  for uri in "${uris[@]}"; do
+    bucket=$(sed -E 's/s3:\/\/([^\/]+).*/\1/' <<< $uri)
+    if [[ ! -z `grep 'hosted-cellxgene' <<< $bucket` ]]; then
+      rdev_bucket_suffix="cellxgene"
+    elif [[ ! -z `grep 'corpora-data' <<< $bucket` ]]; then
+      rdev_bucket_suffix="artifacts"
+    fi
+    key=$(sed -E 's/s3:\/\/([^\/]+)\/(.*)/\2/' <<< $uri)
+    [[ "${key: -1}" == "/" ]] && s3_cmd=$S3_SYNC_CMD || s3_cmd=$S3_COPY_CMD  # use 'sync' if recursive, 'cp' otherwise
+    $s3_cmd $uri s3://env-rdev-${rdev_bucket_suffix}/${STACK}/${key} || echo "$uri does not exist -- skipping..."
+  done
+
+  # Copy public assets
+  dv_id_query="select unnest(datasets) from \"CollectionVersion\" where collection_id in ('$(sed "s/,/','/g" <<< $COLLECTIONS)')"
+  dataset_version_ids=($(PGOPTIONS='-csearch_path=persistence_schema' PGPASSWORD=${DB_PW} psql --dbname corpora_${DEPLOYMENT_STAGE} --username corpora_${DEPLOYMENT_STAGE} --host 0.0.0.0 --csv --tuples-only -c "$dv_id_query"))
+  DEPLOYMENT_STAGE=$DEPLOYMENT_STAGE make db/tunnel/down  # db access no longer needed
+  exts=("rds" "h5ad")
+  for dv_id in "${dataset_version_ids[@]}"; do
+    for ext in "${exts[@]}"; do
+      uri="s3://dataset-assets-public-${DEPLOYMENT_STAGE}/${dv_id}.${ext}"
+      $S3_COPY_CMD $uri s3://env-rdev-datasets/${STACK}/${dv_id}.${ext} || echo "$uri does not exist -- skipping..."
+    done
+  done
+else
+  # TODO: Add --delete once we confirm that is no data in the folders
```

Comment:
```suggestion
  # TODO: Add --delete once we confirm that there is no data in the folders
```
if I may correct my own, original error...